src/detail/sqlite_backend.cc

```diff
@@ -49,11 +49,68 @@ expected<data> from_blob(const void* buf, size_t size) {
     return {ec::invalid_data};
 }
 
+bool optional_enum_option(const broker::backend_options& options,
+                          const std::string& name, const std::string& prefix,
+                          const std::set<std::string>& allowed,
+                          std::string* result) {
+  auto i = options.find(name);
+  if (i == options.end())
+    return true; // no error
+
+  auto value = get_if<broker::enum_value>(i->second);
+  if (!value) {
+    BROKER_ERROR("SQLite backend option '" << name << "' not an enum value");
+    return false;
+  }
+
+  if (value->name.rfind(prefix, 0) != 0) {
+    BROKER_ERROR("SQLite backend option '"
+                 << name << "' not starting with prefix " << prefix);
+    return false;
+  }
+
+  auto sstr = value->name.substr(prefix.size(), std::string::npos);
```

Comment:
No reason to pass in the default explicitly.

```suggestion
  auto sstr = value->name.substr(prefix.size());
```